draft-ietf-tls-esni.md

```diff
@@ -797,9 +806,12 @@ transmitted on the wire by the client:
 1. If CH1 contains the "encrypted_client_hello" extension but CH2 does not, or
    if CH2 contains the "encrypted_client_hello" extension but CH1 does not, then
    the server MUST abort the handshake with an "illegal_parameter" alert.
-1. Suppose the "encrypted_client_hello" extension is sent in both CH1 and CH2,
-   If the configuration identifier (see {{ech-configuration}}) differs between
-   CH1 and CH2, then the server MUST abort with an "illegal_parameter" alert.
+1. If the "encrypted_client_hello" extension is sent in both CH1 and CH2,
+   the server follows the procedure in {{client-facing-server}} to decrypt the
+   second ClientHelloOuter, but it uses the previously-selected ECHConfig as
+   the set of candidate ECHConfigs. If decryption fails, the server aborts the
+   connection with a "decrypt_error" alert rather than continuing the handshake
+   with the second ClientHelloOuter.
```

Comment:
This works for me. However, to align with the new text above, if decryption fails here then the server should ignore ECH and continue with ClientHelloOuter.

Reply:
Right again!